File: day1/day1.go
package day1

import (
	"strconv"
	"strings"
)

func Star1(test string) int {
	frequency := 0
	changes := strings.Split(test, "\n")

	for _, value := range changes {
		currentChange, _ := strconv.Atoi(value)
		frequency += currentChange
	}

	return frequency
}

func Star2(test string) int {
	frequency := 0
	changes := strings.Split(test, "\n")
	pastFrequencyValues := make(map[int]bool)

	for {
		for _, value := range changes {
			currentChange, _ := strconv.Atoi(value)
			frequency += currentChange
			if pastFrequencyValues[frequency] == true {
				return frequency
			}
			pastFrequencyValues[frequency] = true
		}
	}
}

File: day1/day1.py
def star1(test: str) -> int:
    changes = test.split("\n")
    return sum(int(i) for i in changes)


def star2(test: str) -> int:
    changes = test.split("\n")
    frequency = 0
    previousStates = set()
    while True:
        for i in changes:
            frequency += int(i)
            if frequency in previousStates:
                return frequency
            previousStates.add(frequency)
